src/components/SettingsModal.tsx

```diff
@@ -136,6 +183,41 @@ export const SettingsModal = ({ handleCloseModal, isModalOpen }: ISettingsModal)
       ouiaId="settings-modal"
     >
       <Form>
+        {settings.dsl.supportsResourceDescription && (
+          <>
+            <FormGroup
+              fieldId="integration-name"
+              helperTextInvalid="Must be lowercase and alphanumeric (dashes allowed)"
+              validated={nameValidation}
+              isRequired
+              label="Name"
+            >
+              <TextInput
+                aria-describedby="integration-name-helper"
+                data-testid={'settings--integration-name'}
+                id="integration-name"
+                isRequired
+                name="integration-name"
+                onChange={onChangeName}
+                type="text"
+                value={localSettings.name}
+                validated={nameValidation}
+                aria-invalid={nameValidation === 'error'}
+              />
+            </FormGroup>
+            <FormGroup fieldId="integration-description" label="Description">
+              <TextArea
+                aria-describedby="integration-description-helper"
+                data-testid={'settings--integration-description'}
+                id="integration-description"
+                name="integration-description"
+                onChange={onChangeDescription}
+                type="textarea"
+                value={localSettings.description}
+              />
+            </FormGroup>
+          </>
+        )}
```

Comment:
Could we please add a few tests [similar to this one](https://github.com/KaotoIO/kaoto-ui/blob/main/src/components/metadata/MetadataEditorModal.test.tsx#L57) that check:

* Writing in the fields and clicking on save, saves the config to FlowsStore.metadata
* Writing in the fields and clicking on cancel, doesn't do anything (we had a bug in the past on this area that caused changing the fields, even without saving, changed the configuration)

Reply:
Render component in a story
https://github.com/KaotoIO/kaoto-ui/pull/2127/files#diff-6d50e4da11a5a00b9ca77fe1844811c4404bcb6968fed4c51da0269a026cfadfR36
Then use it in the unit test
https://github.com/KaotoIO/kaoto-ui/pull/2127/files#diff-c6b9a444703798b514feefb9878c4c2cc66bea1fd1a74b2648a2380d0b4c186cR52

Reply:
Awesome, thanks for this.